### PR TITLE
feat: handle deep link

### PR DESCRIPTION
**Pull request by johnngugi**

### CODE REVIEW

File: lib/application/core/services/onboarding_utils.dart [hunk 1]
```diff
@@ -382,7 +382,7 @@ OnboardingPathConfig onboardingPath(
 
   if (pinChangeRequired) {
     if (!isPhoneVerified) {
-      return OnboardingPathConfig(BWRoutes.verifySignUpOTP);
+      return OnboardingPathConfig(BWRoutes.phoneLogin);
```
Comment (abiudorina) [inline lib/application/core/services/onboarding_utils.dart:386]: Shouldn't this navigate them to the verify phone page?

File: lib/presentation/core/widgets/handle_deep_link.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    WidgetsBinding.instance?.addPostFrameCallback((_) {
+      StoreProvider.dispatch(context, DeepLinkAction());
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Sentry.captureEvent(
+      SentryEvent(message: const SentryMessage('Handle deep link called')),
+    );
```
Comment (abiudorina) [inline lib/presentation/core/widgets/handle_deep_link.dart:27]: Remove this. I am assuming you were using this to debug?

File: test/widgets_tests/presentation/login/login_page_test.dart [hunk 3]
```diff
@@ -148,22 +149,22 @@ void main() {
...
+      //   }),
+      //   200,
+      // );
+
+      // final MockShortSILGraphQlClient mockShortSILGraphQlClient =
+      //     MockShortSILGraphQlClient.withResponse(
+      //   'idToken',
+      //   'endpoint',
+      //   contentResponse,
+      // );
```
Comment (abiudorina) [inline test/widgets_tests/presentation/login/login_page_test.dart:167]: Remove dead code

**Review by abiudorina — CHANGES_REQUESTED** [review]

File: lib/presentation/core/widgets/handle_deep_link.dart (hunk 2 above)
Comment (johnngugi) [reply lib/presentation/core/widgets/handle_deep_link.dart:27]: Yup

**Review by abiudorina — APPROVED** [review]

**Pull request merged**